ci/build.sh

```diff
@@ -1,166 +1,24 @@
-      (bundle_fluentd_plugin "$(basename "${line}")" "${version}") || exit 1
-    done
-}
-
-function set_up_github() {
-  local token="${1}"
-  local branch="${2}"
-
-  git config --global user.name "Continuous Integration"
-  git remote add origin-repo "https://${token}@github.com/SumoLogic/sumologic-kubernetes-collection.git" > /dev/null 2>&1
-  git fetch --unshallow origin-repo
-
-  readonly branch="$(get_branch_to_checkout)"
-  echo "Checking out the ${branch} branch..."
-  git checkout "${branch}"
-}
-
-function push_docker_image() {
-  local version="$1"
-
-  echo "Tagging docker image ${DOCKER_TAG}:local with ${DOCKER_TAG}:${version}..."
-  docker tag "${DOCKER_TAG}:local" "${DOCKER_TAG}:${version}"
-  echo "Pushing docker image ${DOCKER_TAG}:${version}..."
-  echo "${DOCKER_PASSWORD}" | docker login -u "${DOCKER_USERNAME}" --password-stdin
-  docker push "${DOCKER_TAG}:${version}"
-}
 
-function push_helm_chart() {
-  local version="$1"
-  local chart_dir="$2"
-  local sync_dir="./tmp-helm-sync"
+# shellcheck disable=SC1090
+source "$(dirname "${BASH_SOURCE[0]}")/_build_functions.sh"
 
-  echo "Pushing new Helm Chart release ${version}"
-  set -x
-
-  git checkout -- .
-
-  # due to helm repo index issue: https://github.com/helm/helm/issues/7363
-  # we need to create new package in a different dir, merge the index and move the package back
-  mkdir -p "${sync_dir}"
-  helm package deploy/helm/sumologic --dependency-update --version="${version}" --app-version="${version}" --destination "${sync_dir}"
-
-  git fetch origin-repo
-  git checkout gh-pages
-
-  helm repo index --url "https://sumologic.github.io/sumologic-kubernetes-collection${chart_dir:1}/" --merge "${chart_dir}/index.yaml" "${sync_dir}"
-
-  mv -f "${sync_dir}"/* "${chart_dir}"
-  rmdir "${sync_dir}"
-
-  git add -A
-  git commit -m "Push new Helm Chart release ${version}"
-  git push --quiet origin-repo gh-pages
-  set +x
-}
-
-function build_docker_image() {
-  local tag
-  local no_cache
-  tag="${1}"
-
-  echo "Building docker image with ${tag}:local in $(pwd)..."
-  pushd ./deploy/docker || exit 1
-  cp ../helm/sumologic/conf/setup/main.tf . || exit 1
-  no_cache="--no-cache"
-  if [[ "${DOCKER_USE_CACHE}" == "true" ]]; then
-    no_cache=""
-  fi
-  docker build . -f ./Dockerfile -t "${tag}:local" ${no_cache:+"--no-cache"}
-  rm main.tf
-  rm -f ./gems/*.gem
-  popd || exit 1
+if is_checkout_on_tag; then
+  VERSION="$(git describe --tags --always)"
+  readonly VERSION="${VERSION#v}"
+else
+  # Don't use non-tag 'git describe' output as this will
+  # fail when building fluentd plugins.
+  readonly VERSION="0.0.0"
+fi
 
-  echo "Test docker image locally..."
-  ruby deploy/test/test_docker.rb || exit 1
-}
+: "${DOCKER_TAG:=sumologic/kubernetes-fluentd}"
 
-# Set up Github
-if [ -n "${GITHUB_TOKEN}" ]; then
-  set_up_github "${GITHUB_TOKEN}" "$(get_branch_to_checkout)"
-fi
+echo "Starting build in: $(pwd) with version tag: ${VERSION}"
```

Comment:
Please change `pwd` to the repo main directory. functions won't work otherwise

Reply:
i.e. `cd` to repo root dir or change the `echo`?

Reply:
`cd`/`pushd` to repo root. `echo` is fine

Reply:
we are using in functions relative paths, so if you run the script outside of the repo root it's going to fail

Reply:
Added `pushd "$(dirname "${BASH_SOURCE[0]}")/.." || exit 1`